From c43fb883286d76b453bf03bfbc994b10940ad9f7 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:35 -0400
Subject: [ethernet] i40e: Don't stop driver probe when querying DCB config fails

Message-id: <1409658042-22689-95-git-send-email-sassmann@redhat.com>
Patchwork-id: 90289
O-Subject: [RHEL7.1 PATCH 094/281] i40e: Don't stop driver probe when querying DCB config fails
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Neerav Parikh <Neerav.Parikh@intel.com>

In case of any AQ command to query port's DCB configuration fails
during driver's probe time; the probe fails and returns an error.

This patch prevents this issue by continuing the driver probe even
when an error is returned.

Also, added an error message to dump the AQ error status to show what
error caused the failure to get the DCB configuration from firmware.

Change-ID: Ifd5663512588bca684069bb7d4fb586dd72221af
Signed-off-by: Neerav Parikh <Neerav.Parikh@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 014269ff376f552363ecdab78d3d947fbe2237d9)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 46dc21a..a5ebf42 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -4065,6 +4065,9 @@ static int i40e_init_pf_dcb(struct i40e_pf *pf)
            DCB_CAP_DCBX_VER_IEEE;
    pf->flags |= I40E_FLAG_DCB_ENABLED;
   }
+ } else {
+  dev_info(&pf->pdev->dev, "AQ Querying DCB configuration failed: %d\n",
+    pf->hw.aq.asq_last_status);
  }
 
 out:
@@ -8297,7 +8300,7 @@ static int i40e_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
  if (err) {
   dev_info(&pdev->dev, "init_pf_dcb failed: %d\n", err);
   pf->flags &= ~I40E_FLAG_DCB_ENABLED;
-  goto err_init_dcb;
+  /* Continue without DCB enabled */
  }
 #endif /* CONFIG_I40E_DCB */
 
@@ -8435,9 +8438,6 @@ err_vsis:
 err_switch_setup:
  i40e_reset_interrupt_capability(pf);
  del_timer_sync(&pf->service_timer);
-#ifdef CONFIG_I40E_DCB
-err_init_dcb:
-#endif /* CONFIG_I40E_DCB */
 err_mac_addr:
 err_configure_lan_hmc:
  (void)i40e_shutdown_lan_hmc(hw);
-- 
1.7.1